llvm/lib/Transforms/Scalar/RewriteStatepointsForGC.cpp

```diff
@@ -1249,6 +1249,8 @@ static Value *findBasePointer(Value *I, DefiningValueMapTy &Cache,
 
 #ifndef NDEBUG
   VerifyStates();
+  // get the data layout to compare the sizes of base/derived pointer values
+  auto &DL = cast<llvm::Instruction>(Def)->getModule()->getDataLayout();
```

Comment:
It might be better to move this outside the `ifndef` block and mark it as `[[maybe_unused]]`. It should get optimized away if it isn't used, and it would avoid the possibility of someone seeing it used in the assert and accidentally using it later in the function.